packages/plus/src/components/EstimateCost/EstimateCostProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import deepMerge from 'deepmerge'
+import type { ReactNode } from 'react'
+import { createContext, useCallback, useContext, useMemo } from 'react'
+import EstimateCostLocales from './locales/en'
+
+const EstimateCostContext = createContext<{
+  locales: typeof EstimateCostLocales
+  formatNumber: (number: number, options: FormatNumberOption) => string
+}>({ locales: EstimateCostLocales, formatNumber: () => '' })
+
+export const useEstimateCost = () => useContext(EstimateCostContext)
+
+type EstimateCostProviderProps = {
+  children: ReactNode
+  locales?: typeof EstimateCostLocales
+  currency?: string
+  numberLocales?: string
```

Comment:
i think we can better type this one ?

Reply:
And maybe add a comment for our user where and why it's used ?

Reply:
@philibea 
I took the type from `Intl.NumberFormat` which is a `string`, should I import an external lib for getting all of them?
![Screenshot 2023-11-03 at 10 49 57](https://github.com/scaleway/ultraviolet/assets/15812968/967f5ff9-6c2f-4fc0-9e86-d9eb75107c70)

I already described it into comment on types of each of them, it should be visible when checking JSDoc but also on storybook
![Screenshot 2023-11-03 at 10 51 31](https://github.com/scaleway/ultraviolet/assets/15812968/3e59ce71-c648-4562-81b1-a9f0ce721fdc)